src/app/core/i18n/i18n.service.ts

```diff
@@ -5,41 +5,46 @@ import {loadTranslations} from "@angular/localize";
 @Injectable({
     providedIn: "root",
   })
-  export class I18n {
-    locale = "en";
-    readonly allowedLocales = ["en", "fr", "de", "it"];
+  export class I18nService {
+    readonly allowedLanguages = ["en", "fr", "de", "it"];
     translations: any = {};
-  
-    async setLocale() {
-      const userLocale = localStorage.getItem("locale");
-  
-      // If the user has a preferred language stored in LocalStorage, use it.
-      if (userLocale && this.allowedLocales.includes(userLocale)) {
-        this.locale = userLocale;
-      } else {
-        localStorage.setItem("locale", this.locale);
+    language: string;
+
+    async setLanguage() {
+      const userLanguage = localStorage.getItem("i18nLng");
+      if (userLanguage && this.allowedLanguages.includes(userLanguage)) {
+        this.language = userLanguage;
       }
-  
+      else {
+        const navigatorLanguage = navigator.language.slice(0, 2);
+        if (this.allowedLanguages.includes(navigatorLanguage)) {
+          this.language = navigatorLanguage;
+        } else {
+          this.language = this.allowedLanguages[0];
+        }
+        localStorage.setItem("i18nLng", this.language);
```

Comment:
If the user hasn't explicitly set a language from the menu, it would be best to leave local storage unset. That way, if the user then changes their preferred language in their web browser setting, NGE will take that into account instead of sticking to the old one.

Reply:
I agree, it is changed, thanks for the spot